packages/embark-accounts-manager/src/index.ts

```diff
@@ -64,91 +28,34 @@ export default class AccountsManager {
-      // Check if we have that account in our wallet
-      const account = this.accounts.find((acc) => Web3.utils.toChecksumAddress(acc.address) === Web3.utils.toChecksumAddress(params.reqData.params[0].from));
-      if (account && account.privateKey) {
-        return this.signTransactionQueue.push({payload: params.reqData.params[0], account}, (err: any, newPayload: any) => {
-          if (err) {
-            return callback(err, null);
-          }
-          params.reqData.method = blockchainConstants.transactionMethods.eth_sendRawTransaction;
-          params.reqData.params = [newPayload];
-          callback(err, params);
-        });
-      }
-    }
-    callback(null, params);
-  }
-
-  private async checkBlockchainResponse(params: any, callback: (error: any, result: any) => void) {
-    if (!this.ready) {
-      return callback(null, params);
-    }
-    if ((params.reqData.method === blockchainConstants.transactionMethods.eth_accounts ||
-      params.reqData.method === blockchainConstants.transactionMethods.personal_listAccounts) && this.accounts.length) {
-      if (!arrayEqual(params.respData.result, this.nodeAccounts)) {
-        this.nodeAccounts = params.respData.result;
+  get accounts() {
+    return (async () => {
+      if (!this._accounts) {
         const web3 = await this.web3;
-        this.accounts = AccountParser.parseAccountsConfig(this.embark.config.blockchainConfig.accounts, web3, dappPath(), this.logger, this.nodeAccounts);
+        const nodeAccounts = await web3.eth.getAccounts();
+        this._accounts = AccountParser.parseAccountsConfig(this.embark.config.blockchainConfig.accounts, web3, dappPath(), this.logger, nodeAccounts);
       }
-      params.respData.result = this.accounts.map((acc) => {
-        if (acc.address) {
-          return acc.address;
-        }
-        return acc;
-      });
-      return callback(null, params);
-    }
-    callback(null, params);
+      return this._accounts || [];
+    })();
   }
 
   private async parseAndFundAccounts() {
     const web3 = await this.web3;
+    const accounts = await this.accounts;
 
-    const nodeAccounts = await web3.eth.getAccounts();
-    this.nodeAccounts = nodeAccounts;
-    this.accounts = AccountParser.parseAccountsConfig(this.embark.config.blockchainConfig.accounts, web3, dappPath(), this.logger, nodeAccounts);
-
-    if (!this.accounts.length || !this.embark.config.blockchainConfig.isDev) {
-      this.ready = true;
+    if (!accounts.length || !this.embark.config.blockchainConfig.isDev) {
       return;
     }
     try {
       const coinbase = await web3.eth.getCoinbase();
-      const fundingAccounts = this.accounts
+      const fundingAccounts = accounts
         .filter((account) => account.address)
         .map((account) => {
           return fundAccount(web3, account.address, coinbase, account.hexBalance);
       });
-      await Promise.all(fundingAccounts);
+      await Promise.all([fundingAccounts]);
```

Comment:
```suggestion
      await Promise.all(fundingAccounts);
```

`fundingAccounts` is already an iterable so I it shouldn't wrapped in `[]`. When it's wrapped the Promise will resolve on the next turn of the microtask queue with an array of promises that will likely not have resolved yet.

Reply:
Try this in a Node.js REPL:

```js
p = () => new Promise(resolve => { setTimeout(resolve, 3000) })

x = [p(), p()]; y = Promise.all([x]).then(arr => console.log(arr))
```
You'll see printed:
```
[ [ Promise { <pending>, domain: [Domain] },
    Promise { <pending>, domain: [Domain] } ] ]
```
so the Promise.all resolved on the next turn of the microtask queue, i.e. almost immediately, but the promises in the wrapped array haven't resolved.

compare with:
```js
x = [p(), p()]; y = Promise.all(x).then(arr => console.log(arr))
```
only after ~3 seconds this will be printed:
```
[ undefined, undefined ]
```

Reply:
Thanks for the explanation @michaelsbradleyjr. I fixed this by removing the change and adding one additional null check and the typechecker seems to be happy now (the typechecker was happy without the additional null check as well)... 🤷‍♂️